util/make_new_dif.py

```diff
@@ -143,75 +183,77 @@ def main():
-                                "util/make_new_dif/dif_template.h.tpl")
-
-        with header_template_file.open("r") as f:
-            header_template = Template(f.read())
-
-        header_out_file = dif_dir / "dif_{}.h".format(ip.name_snake)
-        with header_out_file.open("w") as f:
-            f.write(header_template.render(ip=ip))
-
-        print("DIF header successfully written to {}.".format(
-            str(header_out_file)))
-
-    if "autogen" in args.only:
-        # Render all templates
-        for filetype in ["inc", "c", "unittest"]:
-            assert (ip.irqs and "ERROR: this IP generates no interrupts.")
-            # Build input/output file names.
-            if filetype == "unittest":
-                template_file = (
-                    REPO_TOP /
-                    f"util/make_new_dif/dif_autogen_{filetype}.cc.tpl")
-                out_file = (autogen_dif_dir /
-                            f"dif_{ip.name_snake}_autogen_unittest.cc")
-            else:
-                template_file = (
-                    REPO_TOP / f"util/make_new_dif/dif_autogen.{filetype}.tpl")
-                out_file = (autogen_dif_dir /
-                            f"dif_{ip.name_snake}_autogen.{filetype}")
-
-            # Read in template.
-            with template_file.open("r") as f:
-                template = Template(f.read())
-
-            # Generate output file.
-            with out_file.open("w") as f:
-                f.write(template.render(
-                    ip=ip,
-                    irqs=ip.irqs,
-                ))
-
-            # Format autogenerated file with clang-format.
-            assert (
-                shutil.which("clang-format") and
-                "ERROR: clang-format must be installed to format autogenerated"
-                " code to pass OpenTitan CI checks.")
-            try:
-                subprocess.check_call(["clang-format", "-i", out_file])
-            except subprocess.CalledProcessError:
-                logging.error(
-                    f"""failed to format {out_file} with clang-format.""")
-                sys.exit(1)
-
-            print("Autogenerated DIF successfully written to {}.".format(
-                out_file))
-
-    if "checklist" in args.only:
-        checklist_template_file = REPO_TOP / "doc/project/sw_checklist.md.tpl"
-
-        with checklist_template_file.open("r") as f:
-            markdown_template = Template(f.read())
-
-        checklist_out_file = dif_dir / "dif_{}.md".format(ip.name_snake)
-        with checklist_out_file.open("w") as f:
-            f.write(markdown_template.render(ip=ip))
-
-        print("DIF Checklist successfully written to {}.".format(
-            str(checklist_out_file)))
+    for ip in ips:
+        if "header" in args.only:
+            header_template_file = (REPO_TOP /
+                                    "util/make_new_dif/dif_template.h.tpl")
+
+            with header_template_file.open("r") as f:
```

Comment:
If you use pathlib you can do `Path.read_text()` or something like that instead.

Reply:
Fixed.